### DISCUSSION TITLE
How to fill the missing data without including the background area （out of the sample）

### DISCUSSION THREAD

**Discussion by Brian-Liu-rx** [mtex-toolbox/mtex#2150] [Ask Anything]
Hi, everyone,
I need to do the "filling the missing data " to my ebsd file, but it always makes the surface very strangle. How can we fill the missing data without including the background area?

 Thanks!
![image](https://github.com/mtex-toolbox/mtex/assets/69452222/ea145beb-0d8a-4fa3-a4e7-4fd54cb05f58)

**Comment by lfgmorales** [comment]
Hi

Try to gridify the ebsd data before calculating grains

like here, Basic Grain Reconstruction section

cheers

Luiz



<https://mtex-toolbox.github.io/GrainReconstruction.html>
Grain Reconstruction | MTEX<https://mtex-toolbox.github.io/GrainReconstruction.html>
mtex-toolbox.github.io<https://mtex-toolbox.github.io/GrainReconstruction.html>
        [favicon.png] <https://mtex-toolbox.github.io/GrainReconstruction.html>


Luiz F. G. Morales
ETH Zürich
Structural Geology and Tectonics Group
Scientific Center for Optical and Electron Microscopy (ScopeM)
HPM C 57.3
Otto-Stern-Weg 3
8093 Zurich, Switzerland
Phone +41 44 633 37 46
Email: ***@***.***



On 4 Jun 2024, at 12:59, Brian-Liu-rx ***@***.***> wrote:



Hi, everyone,
I need to do the "filling the missing data " to my ebsd file, but it always makes the surface are very strangle. How can we fill the missing data without including the background area?

Thanks!
image.png (view on web)<https://github.com/mtex-toolbox/mtex/assets/69452222/ea145beb-0d8a-4fa3-a4e7-4fd54cb05f58>

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2150>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN4LRX3M6BBNJVEJMM3ZFWMYRAVCNFSM6AAAAABIYIQCY2VHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZWG43TMMZXGQ>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2150", "url": "https://github.com/mtex-toolbox/mtex/discussions/2150", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by Brian-Liu-rx** [reply]
Hi, Luiz, 
    It makes no difference here. Could you check the codes for me ?

    [grain_magb,ebsd.grainId] = calcGrains(ebsd('indexed'),'alpha',3,'threshold',5*degree);
    ebsd_filled = fill(ebsd('indexed'),grain_magb);    

    (In addition, when I delete the (''indexed"), the filling does not work at all.

Brian Liu

**Reply by Brian-Liu-rx** [reply]
@lfgmorales @

**Comment by lfgmorales** [comment]
Hi

Can you upload your data somewhere, the code looks fine to me, I can run a quick example

cheers

Luiz



Luiz F. G. Morales
ETH Zürich
Structural Geology and Tectonics Group
Scientific Center for Optical and Electron Microscopy (ScopeM)
HPM C 57.3
Otto-Stern-Weg 3
8093 Zurich, Switzerland
Phone +41 44 633 37 46
Email: ***@***.***



On 4 Jun 2024, at 14:13, Brian-Liu-rx ***@***.***> wrote:



Hi, Luiz,
It makes no difference here. Could you check the codes for me ?

[grain_magb,ebsd.grainId] = calcGrains(ebsd('indexed'),'alpha',3,'threshold',5*degree);
ebsd_filled = fill(ebsd('indexed'),grain_magb);

(In addition, when I delete the (''indexed"), the filling does not work at all.


Brian Liu

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2150#discussioncomment-9663835>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN735L5PT4H2V6A7J6TZFWVNFAVCNFSM6AAAAABIYIQCY2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TMNRTHAZTK>.
You are receiving this because you commented.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2150#discussioncomment-9663835", "url": "https://github.com/mtex-toolbox/mtex/discussions/2150#discussioncomment-9663835", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by Brian-Liu-rx** [reply]
Hi, Luiz,
Here are the data file! Thanks!
https://github.com/Brian-Liu-rx/my_mtex
Brian Liu

**Reply by Brian-Liu-rx** [reply]
@lfgmorales  Luiz, please check the data.
